### PR TITLE
Use String.Equals instead of String.Compare for equality checks

### PR DESCRIPTION
**Pull request by justinvp**
This PR is a continuation of #298 for the more recently open-sourced libraries.

Per the [Best Practices for Using Strings in the .NET Framework](http://msdn.microsoft.com/en-us/library/dd465121(v=vs.110).aspx):

* Use an overload of the `String.Equals` method to test whether two strings are equal.
* Use the `String.Compare` and `String.CompareTo` methods to sort strings, **not to check for equality**.

Also, in `System.Net.Http` there were uses of `String.ToLowerInvariant().GetHashCode()` and `String.ToUpperInvariant().GetHashCode()` that have been changed to `StringComparer.OrdinalIgnoreCase.GetHashCode()` to avoid the unnecessary string allocation.

Also, in `System.Security.Cryptography.RSA` there were uses of `String.Compare` that didn't specify a `StringComparison`, which means it was doing a culture-sensitive comparison (the default behavior for `String.Compare`). I don't think a culture-sensitive comparison is what was intended here so I changed these to `String.Equals` with `StringComparison.Ordinal`. @bartonjs, let me know if this was intended to be culture-sensitive.

The PR is broken up into separate commits for each area to make it easier to review:

 * `System.IO.FileSystem` /cc @ericstj
 * `System.Net.Http` /cc @davidsh 
 * `System.Security.Cryptography.RSA` and `System.Security.Cryptography.X509Certificates` /cc @bartonjs
 * `System.Xml.ReaderWriter` /cc @krwq 

I can squash the commits after the review, if necessary.

### CODE REVIEW

File: src/System.IO.FileSystem/src/System/IO/Directory.cs [hunk 1]
```diff
@@ -578,12 +578,12 @@ public static void Move(String sourceDirName, String destDirName)
 
             StringComparison pathComparison = PathInternal.GetComparison();
 
-            if (String.Compare(sourcePath, destPath, pathComparison) == 0)
+            if (String.Equals(sourcePath, destPath, pathComparison))
```
Comment (Maxwe11) [inline src/System.IO.FileSystem/src/System/IO/Directory.cs]: change String -> `string`
https://github.com/dotnet/corefx/blob/master/Documentation/coding-style.md (rule 11)

File: src/System.Net.Http/src/System/Net/Headers/AuthenticationHeaderValue.cs [hunk 2]
```diff
@@ -74,19 +74,19 @@ public override bool Equals(object obj)
...
-                    (string.CompareOrdinal(_parameter, other._parameter) == 0);
+                return string.Equals(_scheme, other._scheme, StringComparison.OrdinalIgnoreCase) &&
+                    string.Equals(_parameter, other._parameter, StringComparison.Ordinal);
             }
         }
 
         public override int GetHashCode()
         {
-            int result = _scheme.ToLowerInvariant().GetHashCode();
+            int result = StringComparer.OrdinalIgnoreCase.GetHashCode(_scheme);
```
Comment (stephentoub) [inline src/System.Net.Http/src/System/Net/Headers/AuthenticationHeaderValue.cs]: These changes from using ToLowerInvariant() to using OrdinalIgnoreCase are going to result in different hash codes than were previously being produced.  David should verify that's ok for the library's needs.

**Comment by stephentoub** [comment]
One comment/question, but otherwise it all LGTM.

File: src/System.IO.FileSystem/src/System/IO/Directory.cs (hunk 1 above)
Comment (justinvp) [inline src/System.IO.FileSystem/src/System/IO/Directory.cs]: It also says:

> When editing files, keep new code and changes consistent with the style in the files.

Comment (ericstj) [inline src/System.IO.FileSystem/src/System/IO/Directory.cs]: @justinvp agreed.  We should to a DotNET-bot pass to implement https://github.com/dotnet/corefx/issues/391.

File: src/System.Net.Http/src/System/Net/Headers/AuthenticationHeaderValue.cs (hunk 2 above)
Comment (davidsh) [inline src/System.Net.Http/src/System/Net/Headers/AuthenticationHeaderValue.cs]: Upon reading this:
http://stackoverflow.com/questions/11475737/gethashcode-for-ordinalignorecase-dependent-string-classes

and thinking about this:
"Two unequal objects can have the same hashcode. Though two equal objects should never have different hashcodes. If you use InvariantCulture for your hashcode it will still be correct as far as the contract for Equals goes if it's implemented in terms of OrdinalIgnoreCase."

I think the changes are fine.  If we were worried about serialization and having the hash codes changed like this, that might be a concern.  But the use case here is to support equality checks of the strong-type header classes.

So, bottom line, I'm ok w/ this.

System.Net.Http : LGTM :+1:

**Comment by josteink** [comment]
Wouldn't switching `string.Compare` for `string.Equals` result in introducing case-sensitivity in code which previously had none? Is this change guaranteed to be compatible with existing code on platforms where the file-system structures can be (and often is) treated as case-insensitive?

**Comment by justinvp** [comment]
@josteink, if you look at the implementation in `Directory` and `DirectoryInfo`, you will see that it is passing an explicit `StringComparison` to `Equals` to specify the type of comparison. This is a fully compatible change.

**Comment by bartonjs** [comment]
System.Security LGTM.
